Code/Tools/Libs/GuiFoundation/NodeEditor/Implementation/NodeView.cpp

```diff
@@ -136,9 +136,56 @@ void ezQtNodeView::resizeEvent(QResizeEvent* event)
   UpdateView();
 }
 
+void ezQtNodeView::drawBackground(QPainter* painter, const QRectF& r)
```

Comment:
While it looks nice this unfortunately costs way too much performance and makes the whole node view react very poorly while panning especially when zoomed out a bit. You can leave the code in there but put an if(false) or something like that around it so we can add an editor setting for it in the future.

Reply:
Potentially another option in the graph settings as background grids are fairly standard these days for node tools

Reply:
Do we have knowledge of the zoom level. we could adapt the graph drawing based on zoom lower draw cost when navigating large graphs

Reply:
@C-Core been playing around with this a little. Not the cleanest solution but improves the performance a tone. In future i ideally id want something that has multiple resolutions scales, so at each 0.25 step the graph looks the same but scaled for that resolution if that makes sense

```
  if(m_ViewScale.manhattanLength() > 1.0)
  {
    QPen pfine(plToQtColor(plColorScheme::GetColor(plColorScheme::Black, 9)), 1.0);

    painter->setPen(pfine);
    DrawGrid(painter, 15);
  }

  if(m_ViewScale.manhattanLength() > 0.1)
  {
    double scale = m_ViewScale.manhattanLength() < 0.25 ? 150.0 : 300.0;

    QPen p(plToQtColor(plColorScheme::GetColor(plColorScheme::Gray, 1)), 1.0);

    painter->setPen(p);
    DrawGrid(painter, scale);
  }
  ```